README.md
firefoxos-prebuild
==================

[![GitHub license](https://sinfallas.files.wordpress.com/2016/02/gpl.png)](https://github.com/sinfallas/firefoxos-prebuild/blob/master/LICENSE)
[![DOI](https://zenodo.org/badge/4102/sinfallas/firefoxos-prebuild.svg)](https://zenodo.org/badge/latestdoi/4102/sinfallas/firefoxos-prebuild)
![Status](https://api.travis-ci.org/sinfallas/firefoxos-prebuild.svg) 

scripts to install dependencies to build Firefox OS


changelog
commit 35aaa200070d0f82c4ca4e7ec401f7c03989e178 (HEAD -> master, origin/master, origin/HEAD)
Author: Jesus <sinfallas@gmail.com>
Date:   Sun Jan 31 16:04:39 2021 -0400

    * changelog

commit f8e356ab464a56c02ff642894a4b894ffe6d223e
Author: config <Palencia>
Date:   Mon Jul 6 22:03:27 2020 -0400

    * changelog

commit ce1ccd39d98d58bb7bafa764ff0541613b1d3793
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Apr 27 16:53:47 2020 -0400

    permisos

commit 31dcf9f34bdcfa8285e40734cd135f800c5512f6
Author: jose perez <sinfallas@gmail.com>
Date:   Sun Sep 30 13:28:49 2018 -0400

    * changelog

commit 68f0b0d4323808a390ebd099b74cae2f5844dde8 (tag: v1.0.5)
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Feb 29 21:01:41 2016 -0430

    * changelog

commit a1c379a28ad90a51eb88a0fd7a8076ed37e19b61
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Feb 29 21:01:28 2016 -0430

    * logo licencia

commit 6aebdd110871d443871f793a9a8a246b02ab90f6
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 17 20:29:22 2015 -0430

    * changelog

commit bec4b931880ccfcf96cfcc3e270e1c4ba1157925
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Dec 12 00:16:50 2015 -0430

    * changelog

commit 62defe7a69f4398bcf59ba550d83250b371fc4a4
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 10 22:20:37 2015 -0430

    * changelog

commit fdff3c01d90d65341d5dc564cd2faedfe42a0ad4
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 10 21:20:11 2015 -0430

    * changelog

commit da9fa5d174cac321cb3b5a659c4f8f9d2f420456 (tag: v1.0.4)
Author: jose perez <sinfallas@gmail.com>
Date:   Sun Dec 6 18:05:20 2015 -0430

    * changelog

commit 1bd03aacdb79051c1b1c08c5252074f9282d885d
Author: jose perez <sinfallas@gmail.com>
Date:   Tue Oct 13 21:58:39 2015 -0430

    coveralls

commit 529484fb683a1f698f84db9134e68107803aba6d
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Oct 12 12:46:11 2015 -0430

    licencia

commit 9aed1a9ff20ffbe51b1c67f2dcd2bcddc563c25d
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Aug 10 20:07:38 2015 -0430

    * changelog

commit 14580ce10dd8b37667beac168fedb4783765ce26
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Aug 10 20:07:26 2015 -0430

    * doi

commit 6a7420dda6db6fbfb9811d744e85a09bd138c8e5 (tag: v1.0.3)
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Aug 1 21:29:18 2015 -0430

    * changelog

commit 64a4c2f2f4bf615838ba63b132f91a5a9726c322
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Aug 1 21:29:02 2015 -0430

    * coveralls

commit b0275c0ef4874763c15aa96c1ce80f5a262bc9d1
Author: sinfallas <sinfallas@gmail.com>
Date:   Tue Jan 13 19:19:14 2015 -0430

    * changelog

commit 28720a63cb54eeea5c408b4cca27efab3aac9101
Author: Jose Perez <sinfallas@users.noreply.github.com>
Date:   Mon Jan 12 15:49:11 2015 -0430

    Update 13.10.sh

commit d624e87997e410ef56f108a78cbebc708686de34
Author: Jose Perez <sinfallas@users.noreply.github.com>
Date:   Mon Jan 12 15:48:48 2015 -0430

    Update 13.04.sh

commit 263942c7a297a2afa148900407adf977fadaaa91
Author: Jose Perez <sinfallas@users.noreply.github.com>
Date:   Mon Jan 12 15:48:33 2015 -0430

    Update 12.10.sh

commit cbb40f9f727d7ad5befe7ed4f96096d07c5372c1
Author: Jose Perez <sinfallas@users.noreply.github.com>
Date:   Mon Jan 12 15:48:13 2015 -0430

    Update 12.04.sh

commit 071b2723bd6b898b9c1f4851800b92b8657735ae
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 8 00:48:23 2015 -0430

    * changelog

commit 54d506002fc6aa64ef20c72ac5176065dc171111
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 8 00:47:24 2015 -0430

    bugfix

commit dadf92f49e9646afc2558d31fbb96715133b12a2
Author: sinfallas <sinfallas@gmail.com>
Date:   Wed Dec 3 15:51:51 2014 -0430

    * changelog

commit 8a17e7871a894d7a65e072363778f6aaa7c84577
Author: sinfallas <sinfallas@gmail.com>
Date:   Mon Nov 24 01:04:43 2014 -0430

    * changelog

commit 4720804e0cb1ccbaaba6504aedc49347c1cb790e
Author: sinfallas <sinfallas@gmail.com>
Date:   Mon Nov 24 01:04:30 2014 -0430

    * bugfix

commit 75dfd6db00b103cbb13243030541e98030a2de40
Author: Jose Perez <sinfallas@users.noreply.github.com>
Date:   Wed Nov 19 10:49:56 2014 -0430

    Update README.md

commit 2719e7bb1737f5f4b674548282412e702cdc4504
Author: sinfallas <sinfallas@gmail.com>
Date:   Fri Nov 14 19:39:48 2014 -0430

    * changelog

commit f74d29b46352c4ec96f813d2e6943f4cb8221325
Author: sinfallas <sinfallas@gmail.com>
Date:   Fri Nov 14 19:39:34 2014 -0430

    * se agrega travis

commit f0f14b9286563100002743fc6f527b8f731e7a27 (tag: v1.0.2)
Author: Jesus Palencia <sinfallas@users.noreply.github.com>
Date:   Tue Jun 3 11:44:01 2014 -0430

    Create AUTHORS

commit 40b5568134aebc83f73e4da31072b624f49ac90f
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu May 15 14:08:44 2014 -0430

    * se crea changelog

commit 9af47ea94fbc755b9b3598b1fed1ac26cd68ffe9
Author: sinfallas <sinfallas@gmail.com>
Date:   Tue May 6 21:22:59 2014 -0430

    * Cambios relacionados con el copyright

commit 7933b4773faf57671abefc0cc859cf0d005360ce
Author: sinfallas <sinfallas@gmail.com>
Date:   Sat Apr 26 15:34:54 2014 -0430

    * misc

commit ee42fd8a94c5146f0a7f21a3e4b9b8454587c3aa
Author: sinfallas <sinfallas@gmail.com>
Date:   Sat Apr 26 15:28:45 2014 -0430

    * se agrega informacion de copyright

commit 8469a76da527c7127a81bcaa465be8a7f07d1f91
Author: sinfallas <sinfallas@yahoo.com>
Date:   Sun Mar 30 23:12:04 2014 -0430

    * first commit

commit c91a7163a409aaeb09b66dbda69f58b13dc42bae
Author: Sinfallas <sinfallas@users.noreply.github.com>
Date:   Sun Mar 30 20:40:35 2014 -0700

    Initial commit


13.04.sh
#!/bin/bash
# Ubuntu 13.04
if [[ $USER != root ]]; then
echo "Error: Debe tener privilegios de ROOT"
exit 1
fi
add-apt-repository -y ppa:ubuntu-toolchain-r/test
apt update
apt -y install build-essential libgl1-mesa-dev libglapi-mesa:i386 libgl1-mesa-glx:i386 android-tools-adb
ln -s /usr/lib/i386-linux-gnu/libX11.so.6 /usr/lib/i386-linux-gnu/libX11.so
ln -s /usr/lib/i386-linux-gnu/mesa/libGL.so.1 /usr/lib/i386-linux-gnu/libGL.so
apt-get -y install --no-install-recommends autoconf2.13 bison bzip2 ccache curl flex gawk gcc g++ g++-multilib gcc-4.6 g++-4.6 g++-4.6-multilib git ia32-libs lib32ncurses5-dev lib32z1-dev zlib1g:amd64 zlib1g-dev:amd64 zlib1g:i386 zlib1g-dev:i386 libgl1-mesa-dev libx11-dev make zip gcc-4.7
git clone git://github.com/mozilla-b2g/B2G.git
cd B2G
config.sh emulator
echo "Finalizado"
exit 0


13.10.sh
#!/bin/bash
# Ubuntu 13.10
if [[ $USER != root ]]; then
echo "Error: Debe tener privilegios de ROOT"
exit 1
fi
add-apt-repository -y ppa:ubuntu-toolchain-r/test
dpkg --add-architecture i386
apt update
apt -y install build-essential libgl1-mesa-dev libglapi-mesa:i386 libgl1-mesa-glx:i386 ia32-libs android-tools-adb
ln -s /usr/lib/i386-linux-gnu/libX11.so.6 /usr/lib/i386-linux-gnu/libX11.so
ln -s /usr/lib/i386-linux-gnu/mesa/libGL.so.1 /usr/lib/i386-linux-gnu/libGL.so
apt-get -y install --no-install-recommends autoconf2.13 bison bzip2 ccache curl flex gawk gcc g++ g++-multilib gcc-4.6 g++-4.6 g++-4.6-multilib git lib32ncurses5-dev lib32z1-dev zlib1g:amd64 zlib1g-dev:amd64 zlib1g:i386 zlib1g-dev:i386 libgl1-mesa-dev libx11-dev make zip libxml2-utils gcc-4.7
update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.6 1 
update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 2 
update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.6 1 
update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 2 
update-alternatives --set gcc "/usr/bin/gcc-4.6" 
update-alternatives --set g++ "/usr/bin/g++-4.6" 
git clone git://github.com/mozilla-b2g/B2G.git
cd B2G
config.sh emulator
echo "Finalizado"
exit 0
